### PR TITLE
Toc

### PR DESCRIPTION
**Pull request by winkerVSbecks**
Mostly just porting over to TS and documenting the top level flat list use case. 

The TS post did require quite a bit of refactoring. I want to make sure I didn't introduce any regressions.

### CODE REVIEW

**Review by domyen — APPROVED** [review]

**Review by domyen — CHANGES_REQUESTED** [review]
Left UI feedback in Chromatic

File: src/components/table-of-contents/TableOfContents.tsx [hunk 1]
```diff
@@ -45,29 +50,37 @@ const getOpenState = ({
...
     didChangeCurrentPath,
     depth = 0,
     globalItemUpdate,
     singleItemUpdate,
-    lastFocusedId,
   } = args;
 
   return items.map((item) => {
-    const isMenuWithChildren = item.type === ITEM_TYPES.MENU && !!item.children;
+    const isMenuWithChildren = item.type === 'menu' && !!item.children;
```
Comment (ndelangen) [inline src/components/table-of-contents/TableOfContents.tsx:83]: Why get rid of the constant in favor of a magic string?

Comment (winkerVSbecks) [reply src/components/table-of-contents/TableOfContents.tsx:83]: Good point. I was having some issues with it, but will convert it to enum.

**Comment by winkerVSbecks** [comment]
@domyen @ndelangen could you take another look at this one please

**Review by domyen — APPROVED** [review]

**Review by ndelangen — APPROVED** [review]

**Comment by winkerVSbecks** [comment]
:rocket: PR was released in [`v5.4.0`](https://github.com/storybookjs/design-system/releases/tag/v5.4.0) :rocket:

**Pull request merged**